messages/jsonschema/scripts/templates/java.java.erb

```diff
@@ -0,0 +1,96 @@
+            <%- end -%>
+          <%- end -%>
+        }
+
+      <%- schema['properties'].each do |property_name, property| -%>
+        <%- if (schema['required'] || []).index(property_name) -%>
+        public <%= type_for(class_name(key), property_name, property) -%> get<%= capitalize(property_name) %>() {
+            return java.util.Objects.requireNonNull(<%= property_name %>, "<%= class_name(key) %>.<%= property_name %> cannot be null");
+        }
+
+        public void set<%= capitalize(property_name) %>(<%= type_for(class_name(key), property_name, property) -%> <%= property_name %>) {
+            this.<%= property_name %> = java.util.Objects.requireNonNull(<%= property_name %>, "<%= class_name(key) %>.<%= property_name %> cannot be null");
+        }
+        <%- else -%>
+        public java.util.Optional<<%= type_for(class_name(key), property_name, property) -%>> get<%= capitalize(property_name) %>() {
+            return java.util.Optional.ofNullable(<%= property_name %>);
+        }
+
+        public void set<%= capitalize(property_name) %>(<%= type_for(class_name(key), property_name, property) -%> <%= property_name %>) {
+            this.<%= property_name %> = <%= property_name %>;
+        }
+        <%- end -%>
+
+      <%- end -%>
+        public void validate() {
+        <%- schema['properties'].each do |property_name, property|
+          required = (schema['required'] || []).index(property_name)
+        -%>
+          <%- if required -%>
+            java.util.Objects.requireNonNull(<%= property_name -%>, "<%= class_name(key) %>.<%= property_name %> cannot be null");
+          <%- end -%>
+          <%- if property['type'] == 'array' -%>
+            <%- if property['items']['type'] -%>
+            <% if !required %>if (<%= property_name %> != null) <%- end %><%= property_name %>.forEach(e -> java.util.Objects.requireNonNull(e, "<%= class_name(key) %>.<%= property_name %> elements cannot be null"));
+            <%- else -%>
+            <% if !required %>if (<%= property_name %> != null) <%- end %><%= property_name %>.forEach(<%= type_for(key, nil, property['items']) %>::validate);
+            <%- end -%>
+          <%- elsif property['$ref'] -%>
+            <% if !required %>if (<%= property_name %> != null) <%- end %><%= property_name %>.validate();
+          <%- end -%>
+        <%- end -%>
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            <%= class_name(key) %> that = (<%= class_name(key) %>) o;
+            return <%- schema['properties'].each_with_index do |(property_name, property), index| %>
+                <%- if (schema['required'] || []).index(property_name) -%>
+                <%= property_name -%>.equals(that.<%= property_name -%>)<%= index < schema['properties'].length-1 ? ' && ' : ';' -%>
```

Comment:
With the default constructor present this would result in null pointers.

Reply:
Fields that are nullable are compared with `java.util.Objects.equals`. Non-nullable fields cannot be null because they are verified in `.validate()` (called in the constructor).